src/redux/rockets/reducer.js

```diff
@@ -1,10 +1,17 @@
-import rocketShowed from './action-types';
+import rocketShowed, { rocketBooked } from './action-types';
 
 const startState = [];
 const showRocket = function reducer(state = startState, action = {}) {
   switch (action.type) {
     case rocketShowed:
-      return [...action.payload];
+      return [...state, ...action.payload];
+    case rocketBooked:
+      return state.map((rocket) => {
+        if (rocket.id !== action.payload) {
+          return rocket;
+        }
+        return { ...rocket, reserved: !rocket.reserved };
```

Comment:
Anyways you do not have to worry about mutating the state because map method returns a new array different from the original one 
Thanks for your review @emmiiorji